packages/core/src/initializer/index.ts

```diff
@@ -16,13 +16,34 @@ import { StrykerInquirer } from './stryker-inquirer.js';
 import { createInitializers } from './custom-initializers/index.js';
 import { GitignoreWriter } from './gitignore-writer.js';
 
-const NPM_REGISTRY = 'https://registry.npmjs.com';
+let currentRegistry: string | undefined;
+
+function getRegistry(): string {
```

Comment:
Can you add unit tests for this function? Also, move it to a separate file.

Reply:
I am having some issues with the last case where I would need to use the `execaSync`, I have a few ways I can test this:

**Mocking:**
Would result in not actually testing the functionality of it, wouldn't do anything.

**Editing the `npm config`:**
This may cause a problem if I do `beforeEach` and `afterEach` to edit the `npm config` there are multiple things that could happen:
- If the command fails it can ruin the person's npm configuration.
- If the person aborts mid test it could ruin the person's npm configuration.

Seems like too be too many points of failure.
I have no idea how to test this or if I even should, as I gave a default value if it fails and warned them.
I would wait for your input.

Reply:
Ah yes, great question. Please add 2 more unit tests, 1 for the happy flow and one for the alternate flow using mocking of `execaSync`. You are right that we won't test the actual command `npm config get registry`, but I would expect that to work as documented. However, if you want to add a test for that, you are welcome to add an integration test (see `test/integration`).